### PR TITLE
Remove usage of deprecated node api

### LINKED ISSUE #17: Use of deprecated API
Usage of `new Buffer()` is deprecated. [source](https://nodejs.org/api/deprecations.html#deprecations_dep0005_buffer_constructor)

This outputs a message when used and fails tests when `--throw-deprecation` flag is used, i.e.:
```sh
node --throw-deprecation test/index.js
```

`Buffer.alloc` and `Buffer.from` can be used instead

### PR DESCRIPTION
**Pull request by micburks**
Closes https://github.com/chrisdickinson/bops/issues/17

### CODE REVIEW

File: create.js [hunk 1]
```diff
@@ -2,6 +2,12 @@ module.exports = create
...
 
+var majorVersion = process.version.replace(/^v/, '').split('.')[0];
+
 function create(size) {
-  return Buffer.alloc(size)
+  if (!majorVersion || majorVersion < 6) {
+    return new Buffer(size)
+  } else {
+    return Buffer.alloc(size)
+  }
```
Comment (chrisdickinson) [inline create.js]: Would you mind moving this to the outside of the `create` function? Something like:

```suggestion
var version = ((process || {}).version || 'v0.0.0').slice(1).split('.')[0]
var create = (
  Number(version) < 6
  ? function create (size) { return new Buffer(size) }
  : function create(size) { return Buffer.alloc(size) }
)
```

Comment (micburks) [reply create.js]: 👍

**Review by chrisdickinson — APPROVED** [review]
This looks great! Thank you!

**Comment by chrisdickinson** [comment]
Published as v1.0.1.

**Pull request merged**